library/EventListener/MetadataOperations.php

```diff
@@ -107,8 +109,20 @@ public function search(EventInterface $event) {
             'to' => $params->get('to'),
         ];
 
+        if ($queryParams['page'] < 1) {
+            throw new RuntimeException('Invalid param. "page" must be a positive number.', 400);
+        }
+
+        if ($queryParams['limit'] < 1) {
+            throw new RuntimeException('Invalid param. "limit" must be a positive number.', 400);
+        }
+
         // Parse the query JSON and transform it to an AST
-        $ast = DslParser::parse($metadataQuery);
+        try {
+            $ast = DslParser::parse($metadataQuery);
+        } catch (\Exception $e) {
+            throw new RuntimeException('Invalid metadata query', 400);
```

Comment:
Would it be simpler to change the DslParser to simply throw Runtime Exceptions? Or should it still throw InvalidArgument Exceptions, and then that gets translated into a Runtime Exception? The latter means that any error-message that can help you fix your query disappears which might not be optimal.